src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -737,7 +753,7 @@ private void visitClassOrInterface(String name, FunctionType ftype) {
       // Class extends another class
       if (getSuperType(ftype) != null) {
         emit("extends");
-        visitType(getSuperType(ftype));
+        visitType(getSuperType(ftype), emitInstance);
```

Comment:
I think the only place where we'd ever pass `true` here is for classes extending base classes. In that case, we always know it's a nominal type that we're extending (I don't think you can extend arbitrary type expressions in Closure - can you?).

So I wonder if the code would be cleaner if we'd just directly call the code to emit a nominal type name + possible generic type instantiation here (and also handle the referenced type indirection), instead of calling the more generic `visitType` and then always passing the `emitInstance` flag around when calling `visitType`.

Not sure what's better, but might be worth taking a look.